.github/workflows/dvc-diff.yml

```diff
@@ -0,0 +1,53 @@
+# Checks for image diffs in a Pull Request and adds a GitHub comment showing the diff
+name: DVC image diff
+
+on:
+  pull_request:
+    paths:
+      - 'pygmt/tests/baseline/*.png.dvc'
+
+jobs:
+  dvc-diff:
+    name: DVC image diff
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2.3.4
+      with:
+        # fetch all history so that git diff works
+        fetch-depth: 0
+
+    - name: Setup data version control (DVC)
+      uses: iterative/setup-dvc@v1.0.2
+
+    - name: Setup continuous machine learning (CML)
+      uses: iterative/setup-cml@v1.0.0
+
+    # Produce the markdown diff report, which should look like:
+    # | Status   | Path                                |
+    # |----------|-------------------------------------|
+    # | added    | pygmt/tests/baseline/test_image.png |
+    - name: Get list of images that were added or changed
+      run: |
+        dvc diff --show-md master HEAD >> report.md
+        cat report.md
+
+    - name: Pull image data from cloud storage
+      run: dvc pull --remote upstream
+
+    - name: Report image diff in a GitHub comment
+      env:
+        repo_token: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        # Get just the filename of the changed image from the report
+        tail --lines=+3 report.md | cut --delimiter=' ' --fields=7 > diff_files.txt
+
+        # Append each image to the markdown report
+        echo -e "## Image diff(s)\n" >> report.md
+        while IFS= read -r line; do
+          cml-publish --md "$line" >> report.md
+        done < diff_files.txt
+
+        # Send diff report as GitHub comment
+        cml-send-comment report.md
```

Comment:
In https://github.com/GenericMappingTools/pygmt/pull/1104#issuecomment-804592062, I gave the reasons of why I don't like the new workflow:

> - It makes a new comment for each commit, so the length of the PR page will increase quickly.
> - It comments on both commits and PRs, so we will get two notifications for every commit.

If I understand it correctly, `cml-send-comment` is the command to comment on commits and PRs. I checked the documentation and don't see any options to disable commenting on commits and/or re-use existing comment.

The following steps may work:

- Still use `dvc diff` and `cml-publish` to generate the markdown file
- Use the [`find-comment`](https://github.com/peter-evans/find-comment) action to check if the workflow already created a comment before
- Use the [create-or-update-comment](https://github.com/peter-evans/create-or-update-comment) action to create a new comment or update an old comment, based on the return code of the `find-comment` step.

The [create-or-update-comment](https://github.com/peter-evans/create-or-update-comment) action also provides a good template for us:
```
    - name: Find Comment
      uses: peter-evans/find-comment@v1
      id: fc
      with:
        issue-number: ${{ github.event.pull_request.number }}
        comment-author: 'github-actions[bot]'
        body-includes: This comment was written by a bot!

    - name: Create comment
      if: steps.fc.outputs.comment-id == ''
      uses: peter-evans/create-or-update-comment@v1
      with:
        issue-number: ${{ github.event.pull_request.number }}
        body: |
          This comment was written by a bot!
        reactions: rocket

    - name: Update comment
      if: steps.fc.outputs.comment-id != ''
      uses: peter-evans/create-or-update-comment@v1
      with:
        comment-id: ${{ steps.fc.outputs.comment-id }}
        body: |
          This comment has been updated!
        reactions: hooray
```

Reply:
Cool, thanks for pointing to https://github.com/peter-evans/create-or-update-comment (peter-evans is amazing!), I've updated the workflow so there won't be so many noisy comments anymore as with `cml-publish`. Everything will be in a single comment. See e.g. https://github.com/GenericMappingTools/pygmt/pull/1104#issuecomment-807802379

![image](https://user-images.githubusercontent.com/23487320/112561752-b2909800-8e3a-11eb-96dc-1d5f0a6ab50f.png)